pkg/discovery/dtofactory/service_entity_dto_builder.go

```diff
@@ -76,7 +77,17 @@ func (builder *ServiceEntityDTOBuilder) BuildDTOs() []*proto.EntityDTO {
 
 		ebuilder := sdkbuilder.NewEntityDTOBuilder(proto.EntityDTO_SERVICE, id).
 			DisplayName(displayName)
-
+			// split service name and namespace to set as entity property
+		splitClusterServiceNameAndNameSpace := strings.Split(serviceName, "/")
+		clusterServiceNamespace, clusterServiceName := splitClusterServiceNameAndNameSpace[0], splitClusterServiceNameAndNameSpace[1]
```

Comment:
This splitting of 'serviceName' is redundant and non-optimal.
You have handle to 'service' in line 52 so can easily get the service name and namespace as service.Name and service.Namespace